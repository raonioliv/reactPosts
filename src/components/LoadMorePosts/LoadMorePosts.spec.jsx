import { fireEvent, render, screen } from "@testing-library/react";
import { LoadMorePosts } from "."


describe('<LoadMorePosts />', () =>{
    it('should render the button with text "LOAD MORE"', () => { 
        render(<LoadMorePosts text="Load More" />);
        expect.assertions(1)
        const button = screen.getByRole('button', {name: /load more/i})
        expect(button).toBeInTheDocument()
    })

    it('should trigger the function on click', ()=>{ 
        const fn = jest.fn()
        render(<LoadMorePosts text="Load More" onClick={fn()} />)
        
        const button = screen.getByRole('button', {name: /load more/i})
        fireEvent.click(button)    
        expect(fn).toHaveBeenCalledTimes(1)
    })

    it('should be enabled when its false', ()=>{ 
        render(<LoadMorePosts text="Load More" disabled={false} />)
        
        const button = screen.getByRole('button', {name: /load more/i})
        expect(button).toBeEnabled()
    })

    it('should match the snapshot', ()=>{ 


        const { container } = render(<LoadMorePosts text="Load More" disabled={false} />)

        expect(container.firstChild).toMatchSnapshot();
    })
})