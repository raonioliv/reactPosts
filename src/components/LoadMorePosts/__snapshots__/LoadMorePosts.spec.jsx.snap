// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMorePosts /> should match the snapshot 1`] = `
<div
  class="laodMorePosts__button__wrapper"
>
  <button
    class="loadMorePosts__button"
  >
    Load More
  </button>
</div>
`;
